pkg/controller/dnszone/awsactuator.go

```diff
@@ -433,22 +434,39 @@ func (a *AWSActuator) Delete() error {
 
 func (a *AWSActuator) deleteRecordSets(logger log.FieldLogger) error {
 	logger.Info("Deleting route53 recordsets in hostedzone")
+
+	return DeleteAWSRecordSets(a.awsClient, a.dnsZone, []string{}, logger)
+}
+
+// DeleteAWSRecordSets will clean up all non-NS and non-SOA record types from a DNS zone when no
```

Comment:
The function cleans up NS and SOA records too, just not the ones created with the zone.